dapp/src/components/common/withBridge/index.jsx

```diff
@@ -0,0 +1,176 @@
+  const BridgeWrapper = ({ desiredNetworkName, onConfirmation, ...rest }) => {
+    const { dashboard, network } = useStore()
+    const [transactionHash, setTransactionHash] = useState()
+    const [transactionStatus, setTransactionStatus] = useState()
+    const [confirmationNumber, setConfirmationNumber] = useState()
+    const [receipt, setReceipt] = useState()
+    const [blockNumbers, setBlockNumbers] = useState()
+    const [confirmationsLimit, setConfirmationsLimit] = useState()
+    const [delay, setDelay] = useState(null)
+    const [relayEvent, setRelayEvent] = useState(null)
+    const { homeNetwork, accountAddress } = network
+    const foreignNetwork = dashboard.foreignNetwork
+    const bridgeDirection = dashboard.bridgeDirection ?? 'foreign-to-home'
+    const bridge = dashboard?.bridgeStatus?.from?.bridge
+    const bridgeTo = dashboard?.bridgeStatus?.to?.bridge
+    const bridgeType = dashboard?.community?.bridgeType
+    const homeBridgeAddress = dashboard?.community?.homeBridgeAddress
+    const foreignBridgeAddress = dashboard?.community?.foreignBridgeAddress
+
+    useInterval(() => {
+      if (bridgeTo === 'home') {
+        const web3 = getWeb3({ networkType: homeNetwork })
+        fetchHomeBridgePastEvents({
+          networkType: foreignNetwork,
+          bridgeType,
+          transactionHash,
+          fromBlock: blockNumbers.home,
+          homeBridgeAddress,
+          bridgeDirection,
+        }, { web3 }).then((events) => {
+          const bridgeEvent = getRelayEventByTransactionHash({ events, accountAddress })
+          if (bridgeEvent) {
+            setRelayEvent(bridgeEvent)
+            setDelay(null)
+            onConfirmation()
+          }
+        })
+      } else {
+        const web3 = getWeb3({ networkType: foreignNetwork })
+        fetchForeignBridgePastEvents({
+          networkType: foreignNetwork,
+          bridgeType,
+          fromBlock: blockNumbers.foreign,
+          foreignBridgeAddress,
+          bridgeDirection,
+        }, { web3 }).then((events) => {
+          const bridgeEvent = getRelayEventByTransactionHash({ events, transactionHash, accountAddress })
+          if (bridgeEvent) {
+            setRelayEvent(bridgeEvent)
+            setDelay(null)
+            onConfirmation()
+          }
+        })
+      }
+    }, delay)
+
+    const handleSendTransaction = async (submitType, sendTransaction) => {
+      const promise = sendTransaction()
+      if (submitType === 'transfer') {
+        setConfirmationsLimit(
+          bridge === 'foreign'
+            ? CONFIG.web3.bridge.confirmations.foreign
+            : CONFIG.web3.bridge.confirmations.home
+        )
+        promise.on('confirmation', (confirmationNumber) => {
+          setConfirmationNumber(confirmationNumber)
+        })
+
+        const _web3Home = getWeb3({ networkType: homeNetwork })
+        const _web3Foreign = getWeb3({ networkType: foreignNetwork })
```

Comment:
Can you use web3 instances from the dashboard store here?

Reply:
👍